### ISSUE TITLE
[a11y]: 

### ISSUE THREAD

**Issue by jshenkman** [carbon-design-system/carbon#9896]
### Package

carbon-components

### Browser

Chrome

### Operating System

MacOS

### Package version

v10.42.0

### Automated testing tool and ruleset

IBM equal access accessibility checker

### Description

The tag component container does not have a sufficient contrast ratio with the background for both white and grey 10 theme

### WCAG 2.1 Violation

1.4.3 Contrast (Minimum)

### CodeSandbox example

https://codesandbox.io/s/github/carbon-design-system/carbon/tree/main/packages/react/examples/codesandbox?file=/src/index.js

### Steps to reproduce

Check the contrast ratio for the tag's container background color

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
